src/layers/SoftMaxCrossEntropy.jl

```diff
@@ -52,6 +52,7 @@ function update(l::SoftMaxCrossEntropyLoss, input_size::Tuple;)
 end
 
 function forward(l::SoftMaxCrossEntropyLoss, Y::Array{Float64,2}, label::Array{Int, 2}; kwargs...)
+    # TODO this is a questionable invariant; [label] ranges from 0 to 9 instead of being one hot
```

Comment:
Resolved by bring `SoftMaxCrossEntropy` to its original state; going to resolve this in next pr